### PR TITLE
Close the ChildrenWatch if the node doesn't exist. 

### LINKED ISSUE #149: ChildrenWatch should not log exceptions if the node its watching is deleted
Given the following code:

    import sys
    import logging

    logging.basicConfig(level=logging.DEBUG)

    from time import sleep
    from kazoo.client import KazooClient
    from kazoo.recipe.watchers import ChildrenWatch

    def child_watch(c):
        print "child_watch: {}".format(c)

    zk = KazooClient("127.0.0.1:2181")
    zk.start()

    watch = ChildrenWatch(zk, "/test", child_watch)

    while True:
        sleep(1)
        print ".",
        sys.stdout.flush()

How can I gracefully handle the removal of `/test` from ZooKeeper ? The code raises the following exception, apparently in a watcher thread and I can't act based on that.

I used a `DataWatch` to work around this issue and re-set the `ChildrenWatch` when `/test` reappears, but the presence of the exception makes me think there's a bug/unhandled exception case in the kazoo code.

    INFO:kazoo.client:Received EVENT: Watch(type=4, state=3, path=u'/test')
    INFO:kazoo.client:Sending request(xid=2): GetChildren(path='/test', watcher=<bound method ChildrenWatch._watcher of <kazoo.recipe.watchers.ChildrenWatch object at 0x102271910>>)
    DEBUG:kazoo.client:Reading for header ReplyHeader(xid=2, zxid=103345, err=0)
    INFO:kazoo.client:Received response(xid=2): []
     child_watch: []
    . . . .DEBUG:kazoo.client:Sending request(xid=-2): Ping()
    DEBUG:kazoo.client:Received Ping
     . . .DEBUG:kazoo.client:Sending request(xid=-2): Ping()
    DEBUG:kazoo.client:Received Ping
     .INFO:kazoo.client:Received EVENT: Watch(type=2, state=3, path=u'/test')
    INFO:kazoo.client:Sending request(xid=3): GetChildren(path='/test', watcher=<bound method ChildrenWatch._watcher of <kazoo.recipe.watchers.ChildrenWatch object at 0x102271910>>)
    DEBUG:kazoo.client:Reading for header ReplyHeader(xid=3, zxid=103346, err=-101)
    INFO:kazoo.client:Received error(xid=3) NoNodeError((), {})
    ERROR:kazoo.handlers.threading:Exception in worker queue thread
    Traceback (most recent call last):
      File "/home/user/.pyenv/versions/mac_venv/lib/python2.7/site-packages/kazoo/handlers/threading.py", line 201, in thread_worker
        func()
      File "/home/user/.pyenv/versions/mac_venv/lib/python2.7/site-packages/kazoo/handlers/threading.py", line 289, in <lambda>
        self.callback_queue.put(lambda: callback.func(*callback.args))
      File "/home/user/.pyenv/versions/mac_venv/lib/python2.7/site-packages/kazoo/recipe/watchers.py", line 305, in _watcher
        self._get_children(event)
      File "/home/user/.pyenv/versions/mac_venv/lib/python2.7/site-packages/kazoo/recipe/watchers.py", line 22, in wrapper
        return func(*args, **kwargs)
      File "/home/user/.pyenv/versions/mac_venv/lib/python2.7/site-packages/kazoo/recipe/watchers.py", line 283, in _get_children
        self._path, self._watcher)
      File "/home/user/.pyenv/versions/mac_venv/lib/python2.7/site-packages/kazoo/retry.py", line 122, in __call__
        return func(*args, **kwargs)
      File "/home/user/.pyenv/versions/mac_venv/lib/python2.7/site-packages/kazoo/client.py", line 945, in get_children
        return self.get_children_async(path, watch, include_data).get()
      File "/home/user/.pyenv/versions/mac_venv/lib/python2.7/site-packages/kazoo/handlers/threading.py", line 107, in get
        raise self._exception
    NoNodeError: ((), {})
     . . . .DEBUG:kazoo.client:Sending request(xid=-2): Ping()
    DEBUG:kazoo.client:Received Ping

### PR DESCRIPTION
**Pull request by tonyseek**
This implementation resolves the "NoNodeError" while watching children.

Fix #149.

### CODE REVIEW

File: ensure-zookeeper-env.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@ set -e
 
 HERE=`pwd`
 ZOO_BASE_DIR="$HERE/zookeeper"
-ZOOKEEPER_VERSION=${ZOOKEEPER_VERSION:-3.4.7}
+ZOOKEEPER_VERSION=${ZOOKEEPER_VERSION:-3.4.8}
```
Comment (tonyseek) [inline ensure-zookeeper-env.sh]: ZooKeeper 3.4.7 has been removed from apache.osuosl.org.

**Comment by tonyseek** [comment]
@bbangert Could you review this patch?

File: kazoo/tests/test_watchers.py [hunk 2]
```diff
@@ -171,6 +171,46 @@ def changed(d, stat):
...
+
+        # delete children
+        self.client.delete(path + '/fred')
+        update.wait(3)
+        eq_(args, [[], ['fred'], []])
+        update.clear()
+
+        # delete watching
+        self.client.delete(path)
+        time.sleep(1)
```
Comment (bbangert) [inline kazoo/tests/test_watchers.py]: It'd be nice if there was a non-timing dependent way to wait for this. But I understand there may not be. Maybe if the test tries to grab the watchers _run_lock which should be released as soon as _stopped is set and returned?

**Comment by bbangert** [comment]
LGTM

File: kazoo/tests/test_watchers.py (hunk 2 above)
Comment (tonyseek) [inline kazoo/tests/test_watchers.py]: It is a good idea. I will have a try.

**Pull request merged**